sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/CommunicationTokenCredentialTests.java

```diff
@@ -126,45 +137,92 @@ public void refresherShouldBeCalledImmediatelyWithExpiredToken()
         immediateFresher.resetCallCount();
         CountDownLatch countDownLatch = new CountDownLatch(1);
         immediateFresher.setOnCallReturn(countDownLatch::countDown);
-        CommunicationTokenRefreshOptions tokenRefreshOptions = new CommunicationTokenRefreshOptions(immediateFresher, true, tokenStr);
+        CommunicationTokenRefreshOptions tokenRefreshOptions = new CommunicationTokenRefreshOptions(immediateFresher)
+                .setRefreshProactively(true)
+                .setInitialToken(tokenStr);
+        CommunicationTokenCredential tokenCredential = new CommunicationTokenCredential(tokenRefreshOptions);
+
+        countDownLatch.await();
+        assertEquals(1, immediateFresher.numCalls());
+        StepVerifier.create(tokenCredential.getToken())
+                .assertNext(token ->
+                        assertFalse(token.isExpired(), "Refreshable AccessToken should not expire after refresh"))
+                .verifyComplete();
+        tokenCredential.close();
+    }
+
+    @Test
+    public void fractionalBackoffAppliedWhenTokenExpiring()
+            throws InterruptedException, ExecutionException, IOException {
+        int validForSecs = 8;
+        double expectedTotalCallsTillLastSecond = Math.floor(Math.log(validForSecs));
```

Comment:
I was playing around with log but due to rounding of the epoch timestamp, it was not precise enough. that's why I replaced [this](https://github.com/Azure/azure-sdk-for-js/blob/22acb7535a6c15f6003bbbc4d4a04dafd09ccbeb/sdk/communication/communication-common/test/communicationTokenCredential.spec.ts#L303) with [this](https://github.com/Azure/azure-sdk-for-js/blob/22acb7535a6c15f6003bbbc4d4a04dafd09ccbeb/sdk/communication/communication-common/test/communicationTokenCredential.spec.ts#L28). does it work fine for you?

Reply:
Yeah, it works fine for me cause the code runs till the last second, in your case you are checking till the last ms.